### DISCUSSION TITLE
How to completly disable sampling (cpu, memory and gpu)

### DISCUSSION THREAD

**Discussion by amirouche** [plasma-umass/scalene#235] [Q&A] [answered]
I want to disable sampling completly to have a full view of the memory consumption of a program. How can i do that?

**Comment by emeryberger** [comment, marked as answer]
I'm not quite sure what you mean. Can you elaborate?

**Reply by emeryberger** [reply]
(Copying from thread so it can be marked as an answer)

Scalene uses a non-traditional sampling approach that is guaranteed to never miss memory peaks (at worst, it will underestimate them by just ~1MB).

**Comment by amirouche** [comment]
As far as I understand, with sampling it is possible to miss some peaks of memory. I would like to have a complete picture of the memory consumption of my program.

**Reply by emeryberger** [reply]
Scalene uses a non-traditional sampling approach that is guaranteed to never miss memory peaks (at worst, it will underestimate them by just ~1MB).
